label_studio/ml_models/models.py

```diff
@@ -13,6 +13,12 @@
 logger = logging.getLogger(__name__)
 
 
+# skills are partitions of projects (label config + input columns + output columns) into categories of labeling tasks
+class Skill(models.TextChoices):
```

Comment:
ok, maybe getting ahead of myself, but the more I think about how we'll probably want to handle Skills more broadly, the more I think we'll probably want a separate `Skill` class, proper, and this is more of a `SkillChoice` or `SkillType` or something...
For example, If we do it that way then we can have an object that encodes all of the input classes of a Skill, and can be generated from this with something like `Skill.from_choice(SkillChoice.TEXT_CLASSIFICATION`. I don't love "SkillChoice", so definitely up for a different name, but my feeling is that we'll want to reserve the actual `Skill` class for the actual business logic.
Wdyt?

Reply:
That's reasonable, `SkillName` is a good fit here.

The API sketch I have in mind is something like
```
skill_types = Skill.compatible_with(project.label_config)
skill_name = skill_types[0]
skill = Skill.from_label_config(skill_name, project.label_config)
# or, depending if we use subclassing/mixins/etc
skill_type = skill_types[0]
skill_name = skill_type.name
skill = skill_type.from_label_config(project.label_config)
```